pyethapp/rpc_client.py

```diff
@@ -150,7 +150,7 @@ def nonce(self, address):
         if len(address) == 40:
             address = address.decode('hex')
 
-        res = self.call('eth_getTransactionCount', address_encoder(address), 'pending')
+        res = self.call('eth_nonce', address_encoder(address), 'pending')
```

Comment:
Good point. This is the `rpc_client` of `pyethapp`. I don't know, if anyone considers that a standalone rpc-client. If worst comes to worse, that could be the case.
For `JSONRPCClient.nonce(self)` I wouldn't know how to solve it -- endpoint probing seems overkill. I'll try to add a catch for the missing server method with a reasonable error message. 

For `JSONRPCClient.eth_sendTransaction(self, ...)` `nonce=None` is a valid parameter that get's resolved through the server endpoints (I believe that is how other clients do it).